broker/tasks/cron.py

```diff
@@ -31,15 +32,36 @@
 logger = logging.getLogger(__name__)
 
 
+def get_expiring_certs():
+    query = (
+        select(
+            Certificate.id,
+        )
+        .select_from(Certificate)
+        .join(
+            ServiceInstance,
+            ServiceInstance.current_certificate_id == Certificate.id,
+        )
+        .filter(ServiceInstance.deactivated_at == None)
+        .filter(
+            Certificate.expires_at - datetime.timedelta(days=30)
+            < datetime.datetime.now()
+        )
+    )
+    results = db.session.execute(query).fetchall()
+    certificates = []
+    for result in results:
+        [certificate_id] = result
+        certificate = db.session.get(Certificate, certificate_id)
+        certificates.append(certificate)
+    return certificates
```

Comment:
I think this will fix the problem we're seeing, but we're now running a lot of queries here:
- one query to get the list of certs
- one query _per cert_ to get the cert (line 55)


I think it would be better to have the query return the service instances instead and pass those straight to `instances` on line 65

we're also filtering twice for the service being active (line 45 and line 68) - I think it's an improvement to move that check to 45, but we should drop it on 68 as well

(not your fault, but while you're deleting duplicate checks: we're looking for active operations twice: line 69 and 75. We can get rid of one of those)